src/LeanCode.ContractsGenerator/Compilation/MSBuild/MSBuildHelper.cs

```diff
@@ -16,6 +16,27 @@ public static class MSBuildHelper
 {
     private static readonly string[] RestoreTarget = new string[] { "Restore" };
 
+    private static readonly ImmutableDictionary<string, string> GlobalProperties = ImmutableDictionary.CreateRange(new Dictionary<string, string>
+    {
+        // This property ensures that XAML files will be compiled in the current AppDomain
+        // rather than a separate one. Any tasks isolated in AppDomains or tasks that create
+        // AppDomains will likely not work due to https://github.com/Microsoft/MSBuildLocator/issues/16.
+        ["AlwaysCompileMarkupFilesInSeparateDomain"] = bool.FalseString,
+
+        // Use the preview language version to force the full set of available analyzers to run on the project.
+        ["LangVersion"] = "preview",
+
+#if NET6_0
+        ["TargetFrameworks"] = "net6.0",
+        ["TargetFramework"] = "net6.0",
+#elif NET7_0
+        ["TargetFrameworks"] = "net7.0",
+        ["TargetFramework"] = "net7.0",
```

Comment:
I was hoping we could extract current TFM at runtime but I failed to find API for this :disappointed:

Maybe you could look into this? It could save us the need to update this hack when the next .NET comes out.